### ISSUE TITLE
Plugin request: a 'pay' command that acknowledges internal invoices

### ISSUE THREAD

**Issue by fiatjaf** [renepickhardt/c-lightning-plugin-collection#1]
I'm just leaving this here so maybe someone will find it and create the plugin later, because I don't want to.

Say you generate an invoice, then you later try to pay that invoice on the same node. This `pay` command should succeed with zero fees and that's it.

This is useful for services that hold other people's money and allow then to generate invoices and pay invoices, so they them can pay each other in the same service without requiring complicated code.

**Comment by billygarrison** [comment]
I don't think you can pay your own invoice in c-lightning, at least not currently.

**Comment by fiatjaf** [comment]
That's what the plugin would do: just exclude the invoice and reply with a success message to the `pay` command.

**Comment by billygarrison** [comment]
Would you want something like:

```if senderID == recipientID:
    deleteInvoice
    return "Payment successful."
else:
    pay(bolt11)```

**Comment by fiatjaf** [comment]
Yes.

**Comment by renepickhardt** [comment]
I crated some code for my custodial wallet experiments that was able to pay onself. Can be combined with channel rebalancing
